olive/model/config/kv_cache_config.py

```diff
@@ -0,0 +1,113 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from typing import Dict, Optional
+
+from olive.common.config_utils import ConfigBase
+from olive.common.pydantic_v1 import validator
+
+
+class KVCacheConfig(ConfigBase):
+    ort_past_key_name: str = "past_key_<id>"
+    ort_past_value_name: str = "past_value_<id>"
+    ort_present_key_name: str = "present_key_<id>"
+    ort_present_value_name: str = "present_value_<id>"
+
+    # world_size is used for distributed model. If world_size > 1,
+    # the number of heads should be divisible by world_size
+    world_size: int = 1
+    num_hidden_layers: Optional[int] = None
+    num_attention_heads: Optional[int] = None
+    hidden_size: Optional[int] = None
```

Comment:
make them required.

Reply:
They are required to generate kv cache configs. But I may think they are not to specify by users.

If we make them required here, what if user only want to specify the world_size or other kv_config values in workflow config json? I think that will lead a pydantic parsing error as the auto-filling to these configs needs input model.

Reply:
I think the config is created manually using https://github.com/microsoft/Olive/pull/1121/files#diff-c1d0bb6a2b86e4e881354ced6194029e3da1ca1bf7da8baa1b381e7d63b8331fR123, where we could specify the argument explicitly

Reply:
> where we could specify the argument explicitly.
The line you pointed tells that when `kv_cache` is bool and is true, we create a empty KVCacheConfig. `kv_cache` still can be a dict to let workflows config to parse.

> From kv_cache point of view, the two values are required. If we want deduce, then the creator of kv_cache should handle the deduce logic.
True. I thought that too before. But for the deduction of these values, it needs model's huggingface config that is got during olive runtime. That means it is not easy to use pydantic validator to deduce the value.
Also on another hand, to users, I think these values are not required if they have already give the hugginface model name/class.

Reply:
the deduction of out of scope of pydantic validation. Instead, it is the model responsibility. We could manually check kv_cache PyTorch model. If it is bool, extract all the needed info from model attributes. When all the needed fields are ready, we can use them to construct the KvCacheConfig, In this way, we can split the pydantic validation logic and kv_cache deduction logic. At the same time, the kv_cache will only be type of KvCacheConfig instead of union with bool, which might confuse the end user. 

Basically, the deduction and pydantic validation is two different responsibility and we need split them.

Reply:
Updated.